Makefile

```
@@ -114,6 +114,10 @@ stability-tests: otelcontribcol
 gogci:
 	$(MAKE) $(FOR_GROUP_TARGET) TARGET="gci"
 
+.PHONY: gogenerate
+gogenerate:
```

```suggestion
gogenerate: install-tools
	cd ./internal/tools && go install go.opentelemetry.io/collector/cmd/mdatagen
```

And yes, I'd like to see `mdatagen` be self-contained within the `.tools` dir. That will however require a change to all components in this repository to replace `mdatagen` in the `doc.go` files with `../../.tools/bin/mdatagen` (I think that would be the path). Could do it on this PR but I'd probably recommend opening an issue for that change since it's large.